### PR TITLE
Switch to pyproject/ruff

### PR DESCRIPTION
**Pull request by fwiesel**
Pyproject.toml: PEP 621
So we do not have a bunch of config-files flying around.

Ruff is quite a bit faster and comprehensive than flake

### CODE REVIEW

File: tox.ini
Comment (joker-at-work) [inline tox.ini:1]: Is there an equivalent to `tox.ini` in the `pyproject.toml` or do we just not want `tox` anymore?

Additionally, we might want to add `vcenter-operator` to the `pull-requests` pipeline to do automatic linting :)

Comment (fwiesel) [reply tox.ini:1]: > Is there an equivalent to tox.ini in the pyproject.toml or do we just not want tox anymore?

I think, the ruff config is the equivalent of the removed `tox.ini`

> Additionally, we might want to add vcenter-operator to the pull-requests pipeline to do automatic linting :)

Sure, one thing after the other

**Comment by fwiesel** [comment]
> I assume we do not need "faster" for this code-base, so could you give an example of "comprehensive"?

Faster in this context means, you can have  the linter running online as a [language server](https://github.com/charliermarsh/ruff-lsp), and could get immediate feedback.

Examples for comprehensive are in #28 where I selectively activated some more linters. 
Here is the [compete list](https://beta.ruff.rs/docs/rules/).

**Comment by joker-at-work** [comment]
Ah. English as a second language .. it contains more stuff

File: tox.ini
Comment (fwiesel) [reply tox.ini:1]: I see, I've added a section for tox to call ruff.

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+]
+dependencies = [
+    "attrs",
+    "dnspython @ git+https://github.com/sapcc/dnspython.git@ccloud",
+    "scrypt==0.8.6",
+    "pyvmomi",
+    "jinja2<4,>=3",
+    "kubernetes~=26.1",
+    "keystoneauth1",
+    "dumb-init"]
```
Comment (joker-at-work) [inline pyproject.toml:17]: Why didn't the `cryptography` and `scrypt` dependency make it here? Doesn't `vcenter_operator.masterpassword` still need them?

File: setup.py [hunk 2]
```diff
@@ -1,16 +0,0 @@
...
-# In python < 2.7.4, a lazy loading of package `pbr` will break
-# setuptools if some other modules registered functions in `atexit`.
-# solution from: http://bugs.python.org/issue15881#msg170215
-try:
-    import multiprocessing  # noqa
-except ImportError:
-    pass
-
-setuptools.setup(
-    setup_requires=['pbr'],
```
Comment (joker-at-work) [inline setup.py:14]: Does building via `pyproject.toml` work in the same way as with `pbr`? IIRC, pbr was putting the commit-id into some file after installation, which we use for the changelog in CI.

File: pyproject.toml (hunk 1 above)
Comment (fwiesel) [reply pyproject.toml:17]: Cryptography was only in a comment
```
# cryptography >= 1.6
``` 
as an alternative to `scrypt`, which is still there as the third dependency.

Comment (joker-at-work) [reply pyproject.toml:17]: Looks like I can't read. Sorry :D

**Review by joker-at-work — APPROVED** [review]
I do not see the advantage in the switch to `pyproject.toml`. I don't mind 2 or 3 files instead of having everything in one and would go with less hacks.

That being said, I'm not standing since we have no gap in the functionality.

**Pull request merged**